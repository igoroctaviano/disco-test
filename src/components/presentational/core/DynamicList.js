import React from "react";
import Button from "./Button.js";
import { withLoadingIndicator } from "../../../utils/utilities.js";

const DynamicList = ({
  onLoad,
  buttonText,
  isLoading,
  isButtonActive,
  items = [],
  itemRenderer
}) => {
  const LoadingButton = withLoadingIndicator(Button);
  return items.length > 0 ? (
    <div className="dynamic-list">
      <ul>{items.map(itemRenderer)}</ul>
      {isButtonActive ? (
        <LoadingButton
          onClick={onLoad}
          text={buttonText}
          isLoading={isLoading}
        />
      ) : null}
    </div>
  ) : null;
};

export default DynamicList;
