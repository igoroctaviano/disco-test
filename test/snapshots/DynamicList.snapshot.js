import React from "react";
import DynamicList from "./../../src/components/presentational/core/DynamicList.js";
import renderer from "react-test-renderer";
import { items, itemRenderer } from "./mocks.js";

test("renders correctly", () => {
  const tree = renderer
    .create(<DynamicList items={items} itemRenderer={itemRenderer} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
