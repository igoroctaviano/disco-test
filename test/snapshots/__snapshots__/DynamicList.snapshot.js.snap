// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="dynamic-list"
>
  <ul>
    1
    2
    3
  </ul>
</div>
`;
